### PR TITLE
RF: Strip `AnnotatePaths` from `drop()`

### PR DESCRIPTION
**Pull request by mih**
Ping gh-3368

### CODE REVIEW

File: datalad/distribution/drop.py [hunk 1]
```diff
@@ -178,53 +180,56 @@ def __call__(
...
+
         if dataset and not path:
             # act on the whole dataset if nothing else was specified
             path = refds_path
-        to_drop = []
-        for ap in AnnotatePaths.__call__(
-                dataset=refds_path,
+        content_by_ds = {}
+        for st in Status.__call__(
+                # no not use `ds` to preserve path semantics
```
Comment (kyleam) [inline datalad/distribution/drop.py:201]: s/no/do/ ?

File: datalad/distribution/drop.py [hunk 2]
```diff
@@ -178,53 +180,56 @@ def __call__(
...
                 recursion_limit=recursion_limit,
-                action='drop',
-                # justification for status:
-                # content need not be dropped where there is none
-                unavailable_path_status='notneeded',
-                nondataset_path_status='error',
+                eval_subdataset_state='no',
+                report_filetype='raw',
                 return_type='generator',
-                on_failure='ignore'):
```
Comment (kyleam) [inline datalad/distribution/drop.py:196]: Should `drop` behave the same as `save` here?  It passes `on_failure='ignore'` to `status` and then lets the outer layer decide how to handle non-ok results:

https://github.com/datalad/datalad/blob/a55baf68cbcd624b467b25757bb9689a438e15d4/datalad/core/local/save.py#L208-L216

**Review by kyleam — COMMENTED** [review]
Nice.  

I think the `test_get_cached_dataset` failure in one of the Travis builds is unrelated.

https://travis-ci.org/github/datalad/datalad/jobs/734784386#L2249

File: datalad/distribution/drop.py (hunk 1 above)
Comment (mih) [reply datalad/distribution/drop.py:201]: Yes, thx, fixed!

File: datalad/distribution/drop.py (hunk 2 above)
Comment (mih) [reply datalad/distribution/drop.py:196]: Oh, indeed! Thanks for catching that.

**Pull request merged**